test/CodeGenerator/InjectorGeneratorTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-di for the canonical source repository
+ * @copyright Copyright (c) 2017 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   https://github.com/zendframework/zend-di/blob/master/LICENSE.md New BSD License
+ */
+
+namespace ZendTest\Di\CodeGenerator;
+
+use PHPUnit\Framework\TestCase;
+use Zend\Di\CodeGenerator\InjectorGenerator;
+use Zend\Di\Config;
+use Zend\Di\Definition\RuntimeDefinition;
+use Zend\Di\Resolver\DependencyResolver;
+use ZendTest\Di\TestAsset;
+use Zend\Di\InjectorInterface;
+
+/**
+ * FactoryGenerator test case.
+ */
+class InjectorGeneratorTest extends TestCase
+{
+    const DEFAULT_NAMESPACE = 'ZendTest\Di\Generated';
+
+    use GeneratorTestTrait;
+
+    public function testGenerateCreatesFiles()
+    {
+        $config = new Config();
+        $resolver = new DependencyResolver(new RuntimeDefinition(), $config);
+        $generator = new InjectorGenerator($config, $resolver, self::DEFAULT_NAMESPACE);
+
+        $generator->setOutputDirectory($this->dir);
+        $generator->generate([
+            TestAsset\RequiresA::class
+        ]);
```

Comment:
If this is generating actual files in the filesystem, we should be using [vfsStream](https://github.com/mikey179/vfsStream) to abstract the filesystem. Doing so prevents the need for cleanup after the tests are complete, and will prevent any of these generated files from accidentally getting committed to the repository.

Reply:
Valid point. Is ist possible to include these files then? Im planning to test if the code can be loaded and works as expected in the next steps.

Reply:
That I'm less sure about. :) Give it a try and see!

Another possibility is to check the structure of those files, or even compare them against canned expectations.

Reply:
I will, but not today. It's late here and this is a potential for more typos ;-)

Reply:
> That I'm less sure about. :) Give it a try and see!

It did work quite well :-)

Reply:
\o/